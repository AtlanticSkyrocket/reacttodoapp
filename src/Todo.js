import React from 'react';


function Todo({id, removeTodo, todo}) {
  const handleRemove = evt => {
    const { id } = evt.target.parentElement;
    removeTodo(id);
  }
  return (
  <div className="Todo-container" id={id}>
    <p>{todo}</p>
    <button className="Remove-btn" onClick={handleRemove}>X</button>
  </div>
  )
}

export default Todo;