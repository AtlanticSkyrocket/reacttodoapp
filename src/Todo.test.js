import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Todo from './Todo';


const mockRemoveTodo = jest.fn();

describe('Todo Component', () => {
  it('renders todo correctly', () => {
    render(<Todo id={1} removeTodo={mockRemoveTodo} todo="Test Todo" />);
    
    expect(screen.getByText('Test Todo')).toBeInTheDocument();
  });

  it('snapshot matches', () => {
    const { asFragment } = render(
      <Todo id={1} removeTodo={mockRemoveTodo} todo="Test Todo" />
    );
    expect(asFragment).toMatchSnapshot();
  });

  it('calls removeTodo when the remove button is clicked', () => {
    render(<Todo id={1} removeTodo={mockRemoveTodo} todo="Test Todo" />);
    
    fireEvent.click(screen.getByText('X'));
    
    expect(mockRemoveTodo).toHaveBeenCalledWith('1');
  });
});