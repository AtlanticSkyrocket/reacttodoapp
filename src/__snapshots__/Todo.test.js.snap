// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Component snapshot matches 1`] = `[Function]`;
